src/tab/Tab.tsx

```diff
@@ -6,15 +6,15 @@ import classNames from "classnames";
 import TabHeaderItem from "./header/item/TabHeaderItem";
 import List from "../list/List";
 
-export type TabItem = {
-  id: string;
+export type TabItem<ID = string | number> = {
```

Comment:
In this version, we can pass boolean or false to the generic type. We can even pass array :D Should we? IMO, we should be able to pass string or number only.
```suggestion
export type TabItem<ID extends string | number = string | number> = {
```

Reply:
You're right. I missed it and gave default types only, I should have limited it using constraints. 
Thank you for noticing ✌️